py/embedded-server/deephaven_server/__init__.py

```diff
@@ -14,3 +19,55 @@
 # distribution name from the package name.
 # https://docs.python.org/3/library/importlib.metadata.html#package-distributions
 __version__ = importlib.metadata.version('deephaven-server')
+
+
+def main():
+    """
+    Main entry point for the Deephaven embedded server application.
+    Accepts a command to start the application and the arguments to use.
+    Defaults to `start`, which starts the server.
+    """
+    parser = argparse.ArgumentParser(description="Start the Deephaven embedded server")
+    parser.add_argument("cmd", nargs="?", default="start", choices=["start"], help="Deephaven command to run. Defaults to start.")
+    parser.add_argument("-p", "--port", default=8080, type=int, help="The port to bind to. Defaults to 8080.")
+    parser.add_argument("--host", default="localhost", help="The host to bind to. Defaults to localhost.")
+    parser.add_argument("--key", default=None, help="The key to use. Uses a randomly generated key if not specified.")
+    parser.add_argument("--jvm-args", default=None, help="The JVM arguments to use.")
+    parser.add_argument("--dh-args", default=None, help="The Deephaven arguments to use.")
+    args = parser.parse_args()
+
+    if args.cmd == "start":
```

Comment:
If you step back and think of where this script should go, "start" seems awkward.  It seems like you would want:
* Fresh IDE (--ide option which is default?)
* Fresh Python CLI (--no-ide option?)
* Run a script and then get an IDE (maybe you get this if you have a script ARG with the default --ide)
* Run batch script (maybe you get this if you have a script ARG and --no-ide)

Reply:
Ah yea starting a console would be a good idea too. So:
```
deephaven                            # Aliases to `deephaven start`
deephaven start                    # Opens the Web UI in the browser
deephaven start --console    # Opens a python CLI instead of web UI
deephaven run <file>            # Run a specific file, can use the `--console` flag
```
If we default to `--ide`, doesn't make sense to have it as a flag (IMO), and then having a negative flag `--no-ide` seems odd.
Let me know what you think.

Reply:
Look at how click does boolean flags, and my comment will make more sense.
https://click.palletsprojects.com/en/8.1.x/options/#boolean-flags

Reply:
I think we could also consider having the `deephaven-core` wheel own the main `deephaven` entrypoint, as I imagine we _could_ use it in other contexts. When `deephaven-server` is in the environment, it could add-in `deephaven server` subcommand. I think some of these would be useful even if the user is running the native application workflow (ie, _not_ `deephaven-server`).

```sh
deephaven config show
deephaven plugins list
deephaven server start
```
etc

Reply:
Yea that makes sense... it could also have the server start print out a message if `deephaven_server` isn't installed, something along the lines of:

```
def server():
  """
  Run a deephaven_server command
  """
  try:
    import deephaven_server
  except Exception as e:
    print("You need to install the `deephaven_server` package")
    sys.exit(1)
  ... # process the command
```